activerecord/lib/active_record/core.rb

```diff
@@ -583,5 +580,11 @@ def thaw
       def custom_inspect_method_defined?
         self.class.instance_method(:inspect).owner != ActiveRecord::Base.instance_method(:inspect).owner
       end
+
+      def filter_attribute?(attribute_name)
+        filter_attributes = self.filter_attributes.map(&:to_s).to_set
```

Comment:
We're now doing this work for each attribute, instead of reusing it. I don't think making the implementation slower to avoid some duplication is a good tradeoff.

We could build the string set when `filter_attributes` is assigned, which would allow it to be reused across multiple calls to `#inspect` or `#pretty_print`.

Reply:
Ugh, I didn't notice that it would work slower. Thanks for the review.